engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/graph/ScopeImpl.scala

```diff
@@ -195,23 +237,12 @@ sealed private[graph] class ScopeImpl(
     */
   private[analyse] def resolveUsage(
     occurrence: GraphOccurrence.Use,
-    hint: GraphOccurrence.Def,
     parentCounter: Int = 0
   ): Option[Graph.Link] = {
-    val definition =
-      if (hint != null && (hint.scope() eq this)) {
-        Some(hint)
-      } else {
-        occurrences.values.find {
-          case GraphOccurrence.Def(_, name, _, _, _) =>
```

Comment:
This iteration over `occurrences` was slowing things done. Replaced by `defsBySymbol.get`